internal/component/otelcol/internal/livedebuggingconsumer/livedebuggingconsumer.go

```diff
@@ -33,6 +40,25 @@ func New(debugDataPublisher livedebugging.DebugDataPublisher, componentID string
 	}
 }
 
+// SetTargetConsumers stores the componentIDs of the next consumers
+func (c *Consumer) SetTargetConsumers(metric, log, trace []otelcol.Consumer) {
+	c.mut.Lock()
+	defer c.mut.Unlock()
+	c.targetComponentIDsMetric = extractIds(metric)
+	c.targetComponentIDsLog = extractIds(log)
+	c.targetComponentIDsTraces = extractIds(trace)
+}
+
+func extractIds(consumers []otelcol.Consumer) []string {
+	ids := make([]string, 0, len(consumers))
+	for _, cons := range consumers {
+		if lazy, ok := cons.(*lazyconsumer.Consumer); ok {
+			ids = append(ids, lazy.ComponentID())
+		}
```

Comment:
This is fragile too IMO because next time someone refactors and adds another layer of consumer wrappers for some reason, this will stop working

Can we avoid having to type cast entirely?